index.js

```diff
@@ -1,63 +1,271 @@
+  var i = fc.features.length;
+  while (i--) {
+    if (!filterFn(fc.features[i].properties.netWinding)) {
+        fc.features.splice(i, 1);
+    } else {
+      fc.features[i].properties = {};
+    }
+  }
+  return fc;
+}
+
+function unionFeatureCollection(fc) {
+  // Note: union takes a polygon, but return a polygon or multipolygon (which it can not take in). In case of buffes, however, it will always return a polygon
+  if (fc.features.length == 0) return {type: "Feature", geometry: null};
+  var incrementalUnion = fc.features[0];
+  if (fc.features.length == 1) return incrementalUnion;
+  for (var i = 1; i < fc.features.length; i++) {
+    incrementalUnion = union(incrementalUnion, fc.features[i]);
+  }
+  return incrementalUnion
+}
+
+function offsetToBuffer(polygonOffset) {
+  var unionWithWindingOne = unionFeatureCollection(filterNetWinding(simplepolygon(polygonOffset), function (netWinding){return netWinding == 1}));
+  var unionWithWindingZero = unionFeatureCollection(filterNetWinding(simplepolygon(polygonOffset), function (netWinding){return netWinding == 0}));
+  // This last one might have winding -1, so we might have to rewind it if the difference algorithm requires so
+
+  if (unionWithWindingOne.geometry == null) return {type: "Feature", geometry: null};
+  if (unionWithWindingZero.geometry == null) return unionWithWindingOne;
+  return difference(unionWithWindingOne, unionWithWindingZero);
+}
+
+// This function awaits possible future use
+function winding(poly){
+  // compute winding of first ring
+  var coords = poly.geometry.coordinates[0];
+  var leftVtx = 0;
+  for (var i = 0; i < coords.length-1; i++) { if (coords[i][0] < coords[leftVtx][0]) leftVtx = i; }
+  return (coords[(leftVtx-1).mod(coords.length-1)][1] > coords[(leftVtx+1).mod(coords.length-1)][1]) ? 1 : -1;
+}
+
+// This function awaits possible future use
+function rewind(poly){
+  // outer ring to winding +1, inner rings to winding -1
+  if (winding(helpers.polygon([poly.geometry.coordinates[0]])) == -1) poly.geometry.coordinates[0] = poly.geometry.coordinates[0].reverse();
+  for (var i = 1; i < poly.geometry.coordinates.length; i++) {
+    if (winding(helpers.polygon([poly.geometry.coordinates[i]])) == 1) poly.geometry.coordinates[i] = poly.geometry.coordinates[i].reverse();
+  }
+  return poly
+}
+
+function removeDuplicates(arr) {
+  for (var i = arr.length-1; i > 0; i--) {
+    if (arr[i].equals(arr[i-1])) {
+      arr.splice(i,1);
+    }
+  }
+  return arr;
+}
+
+
+// Function to compare Arrays of numbers. From http://stackoverflow.com/questions/7837456/how-to-compare-arrays-in-javascript
+// Warn if overriding existing method
+// if(Array.prototype.equals) console.warn("Overriding existing Array.prototype.equals. Possible causes: New API defines the method, there's a framework conflict or you've got double inclusions in your code.");
+// attach the .equals method to Array's prototype to call it on any array
+Array.prototype.equals = function (array) {
```

Comment:
Per the JavaScript norms currently, [extending built-in objects can be dangerous](http://perfectionkills.com/whats-wrong-with-extending-the-dom/), since the extensions may break other code or native behavior. Could this be implemented instead as a function that takes two arrays, and the same with the `mod` extension?